homeassistant/components/websocket_api/http.py

```diff
@@ -161,11 +163,31 @@ async def _writer(self) -> None:
                     messages.append(process if isinstance(process, str) else process())
                     messages_remaining -= 1
 
-                joined_messages = ",".join(messages)
-                coalesced_messages = f"[{joined_messages}]"
-                if debug_enabled:
-                    debug("%s: Sending %s", self.description, coalesced_messages)
-                await send_str(coalesced_messages)
+                coalesced_messages = f"[{','.join(messages)}]"
+                # We only check the length of the coalesced messages after
+                # combining them to avoid the overhead of checking the length
+                # of each message individually, and since the majority of
+                # the time we will never hit the max message size limit this
+                # is a good tradeoff.
+                if len(coalesced_messages) < max_message_size:
+                    if debug_enabled:
+                        debug("%s: Sending %s", self.description, coalesced_messages)
+                    await send_str(coalesced_messages)
+                    continue
+
+                logger.warning(
```

Comment:
I'm not sure a warning is warranted here, what's the user supposed to do?
Would `info` be better?

Reply:
Switched to info